lib/modules/dosomething/dosomething_fact/dosomething_fact.module

```diff
@@ -77,6 +77,110 @@ function dosomething_fact_get_mutiple_fact_field_vars($node, $field_names) {
 }
 
 /**
+ * Returns data for multiple specified fields.
+ *
+ * @param object $node The loaded node to collect field data from.
+ * @param array $field_names Array of field machine names to collect data from.
+ * @return array|null
+ */
+function dosomething_fact_get_facts_data($node, $field_names) {
+  $entity = NULL;
+  $facts = [];
+  $sources = [];
+  $language = dosomething_global_convert_country_to_language(dosomething_global_get_current_prefix());
+
+  foreach($field_names as $field_name) {
+    $field_data = $node->$field_name;
+
+    if ($field_data) {
+      if (array_key_exists($language, $field_data)) {
+        $entity = $field_data[$language][0]['entity'];
+      }
+
+      if (isset($field_data[0]['entity']) && is_object($field_data[0]['entity'])) {
+        $entity = $field_data[0]['entity'];
+      }
+    }
+
+    if ($entity) {
+      $entity_data = dosomething_fact_get_entity_data($entity);
+
+      $facts[$field_name]['fact'] = $entity_data['title'];
+      $facts[$field_name]['nid'] = $entity_data['nid'];
+      $fact_sources = dosomething_helpers_extract_field_data($entity->field_source_copy);
```

Comment:
Yeah, so at the point where this is called `dosomething_helpers_extract_field_data()` in that segment of code, the lang assigned to the data is `und` so doesn't really play a role here. This is for the fact sources which are assigned to the Fact entity and would potentially be translated, but wouldn't end up having a language assigned to them... just the parent Fact entity.